etl/exporter/record_exporter_server_integration_test.go

```diff
@@ -0,0 +1,210 @@
+// Copyright 2020 Insolar Network Ltd.
+// All rights reserved.
+// This material is licensed under the Insolar License version 1.0,
+// available at https://github.com/insolar/block-explorer/blob/master/LICENSE.md.
+
+// +build integration
+
+package exporter
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"testing"
+
+	"github.com/insolar/insolar/insolar"
+	"github.com/insolar/insolar/insolar/gen"
+	"github.com/jinzhu/gorm"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+
+	"github.com/insolar/block-explorer/configuration"
+	"github.com/insolar/block-explorer/etl/models"
+	"github.com/insolar/block-explorer/etl/storage"
+	"github.com/insolar/block-explorer/instrumentation/belogger"
+	"github.com/insolar/block-explorer/testutils"
+)
+
+var getRecordGRPCConnection *grpc.ClientConn
+var testDB *gorm.DB
+
+func TestMain(m *testing.M) {
+	// create storage
+	ctx := context.Background()
+	port, err := getFreePort()
+	if err != nil {
+		panic(err)
+	}
+	var dbCleaner func()
+	testDB, dbCleaner, err = testutils.SetupDB()
+	if err != nil {
+		belogger.FromContext(context.Background()).Fatal(err)
+	}
+
+	cfg := configuration.Exporter{
+		Listen: fmt.Sprintf(":%d", port),
+	}
+	s := storage.NewStorage(testDB)
+
+	// create grpc server
+	recordExporter := NewRecordServer(ctx, s, cfg)
+	grpcServer := grpc.NewServer()
+	RegisterRecordExporterServer(grpcServer, recordExporter)
+	exporterServer := NewServer(cfg.Listen, grpcServer)
+	err = exporterServer.Start(ctx)
+	if err != nil {
+		panic(err)
+	}
+	defer func() {
+		exporterServer.Stop(ctx)
+		grpcServer.GracefulStop()
+		dbCleaner()
+	}()
+
+	// create client
+	serverAddr := fmt.Sprintf("localhost:%d", port)
+	var opts []grpc.DialOption
+	opts = append(opts, grpc.WithInsecure(), grpc.WithBlock())
+	getRecordGRPCConnection, err = grpc.Dial(serverAddr, opts...)
+	if err != nil {
+		panic(err)
+	}
+	defer getRecordGRPCConnection.Close()
+	retCode := m.Run()
+	os.Exit(retCode)
+}
+
+func TestRecordServer_GetRecords(t *testing.T) {
+	var accountPrototypeReference, _ = insolar.NewObjectReferenceFromString("insolar:0AAABAiGN1L8F9gCH_keBaxOP4atp9fzLiIci7xOg-hs")
```

Comment:
would be great to add a test case with search by several prototypes, now I see only one PrototypeReference